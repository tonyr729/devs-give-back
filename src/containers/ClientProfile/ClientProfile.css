.header {
  height: 20vh;
  width: 100vw;
  background: linear-gradient(#96c0d6, #536c79); }

.header-container {
  height: 100%;
  width: 64rem;
  display: block;
  margin: 0 auto;
  background-color: #607d8b; }
  .header-container img {
    border-radius: 3rem;
    width: 9rem;
    margin: 3rem 2rem 0; }
  .header-container p {
    font-size: 2rem;
    text-align: -webkit-auto;
    position: relative;
    font-weight: 100;
    top: 7rem;
    left: 8rem; }
  .header-container ul {
    list-style: none; }
  .header-container li {
    display: inline;
    margin-right: 1rem;
    font-size: .8rem;
    color: #f8f8ff;
    position: relative;
    top: 7.5rem;
    left: 9.9rem; }

.problem {
  height: 40vh;
  width: 100vw;
  background: radial-gradient(#000, #181717); }

.problem-container {
  height: 100%;
  width: 64rem;
  display: block;
  margin: 0 auto;
  background-color: black; }

.section-title {
  font-size: 3rem;
  padding: 1rem; }

.problem-title {
  font-size: 2rem;
  padding: 4rem 6rem 0;
  text-align: -webkit-auto; }

.problem-body {
  font-size: 1rem;
  width: 50rem;
  text-align: -webkit-auto;
  font-weight: 100; }

.stats {
  height: 40vh;
  width: 100vw;
  background: linear-gradient(#0d4d88, #1471c9); }

.stats-container {
  height: 100%;
  width: 64rem;
  display: block;
  margin: 0 auto;
  background-color: #1883e7; }

.stats-title {
  font-size: 2rem;
  padding: 4rem 6rem 0;
  text-align: -webkit-auto; }
