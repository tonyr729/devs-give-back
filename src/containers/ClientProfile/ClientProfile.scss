.header {
  height: 20vh;
  width: 100vw;
  // background-color: #536c79;
  background: linear-gradient(#96c0d6, #536c79);

}

.header-container {
  height:100%;
  width:64rem;
  display: block;
  margin: 0 auto;
  background-color: #607d8b;
  img {
    border-radius: 3rem;
    width: 9rem;
    margin: 3rem 2rem 0;
  };
  p {
    font-size: 2rem; 
    text-align: -webkit-auto; 
    position: relative; 
    font-weight: 100;
    top: 7rem; 
    left: 8rem; 
  };
  ul {
    list-style: none;
  };
  li {
    display: inline;
    margin-right: 1rem;
    font-size: .8rem;
    color: #f8f8ff;
    position: relative;
    top: 7.5rem;
    left: 9.9rem;
  };
}

.problem {
  height: 40vh;
  width: 100vw;
  // background-color: rgba(0, 0, 0, 0.712);
  background: radial-gradient(#000,#181717);
}

.problem-container {
  height:100%;
  width:64rem;
  display: block;
  margin: 0 auto;
  background-color: rgb(0, 0, 0)
}

.section-title {
  font-size: 3rem;
  padding-top: 2rem;
}

.problem-title {
  font-size: 1.6rem;
  padding: 4rem 6rem 0;
  text-align: -webkit-auto;
}

.problem-body {
  font-size: 1rem;
  width: 50rem;
  text-align: left;
  color: grey;
  margin: 0 0 0 6.2rem;
  font-weight: 100;
}

.stats {
  height: 40vh;
  width: 100vw;
  // background-color: #1471c9;
  background: linear-gradient(#0d4d88,#1471c9);
}

.stats-container {
  height:100%;
  width:64rem;
  display: block;
  margin: 0 auto;
  background-color: rgb(24, 131, 231);
  
}

.stats-title {
  font-size: 2rem;
  padding: 4rem 6rem 0;
  text-align: -webkit-auto;
}
