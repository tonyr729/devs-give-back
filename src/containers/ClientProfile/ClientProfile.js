import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';

import DatabaseHelper from '../../helpers/DatabaseHelper';
import DataCleaner from '../../helpers/DataCleaner';
import APIHelper from '../../helpers/APIHelper';
import { addRepoStats, signInClient } from '../../actions/actions';
import handsPhoto from '../../images/hands.svg';
import linesPhoto from '../../images/code.svg';
import updatesPhoto from '../../images/update.svg';
import hourPhoto from '../../images/time.svg';

import './ClientProfile.css';

export class ClientProfile extends Component {
  constructor() {
    super();
    this.state = {
      problem: ''
    };
    this.database = new DatabaseHelper();
    this.cleaner = new DataCleaner();
    this.api = new APIHelper();
  }
  
  async componentDidMount() {
    const stats = await this.getRepoStats(this.props.clientsProblem);
    if (stats){
      await this.props.addRepoStats(stats);
    }
  }
  
  getRepoStats = async (problem) => {
    let stats = null;
    if (problem && problem.dev) {

      const repoDev = Object.values(problem.dev).find(dev => dev.repo);
      const repoURL = repoDev.repo;
      console.log(repoURL);
      const apiURL = this.cleaner.cleanRepoURL(repoURL);
      try {
        const lines = await this.api.fetchLinesOfCode(apiURL);
        const contributers = await this.api.fetchNumberOfContributers(apiURL);
        const updates = await this.api.fetchNumberOfUpdates(apiURL);
        const hours = await this.api.fetchNumberOfHours(apiURL);
        stats = { lines, contributers, updates, hours };
      } catch (error) {
        console.log(error)
        this.props.history.push('/error-page');
      }
    }
    return stats;
  }

  displayStats = (stats) => {
    if (stats){
      const contributer = stats.contributer > 1 ? 'contributer' : 'contributers';
      return (
        <div className="stats-container">
          <div className="stats-card">
            <img src={handsPhoto} alt="illustration of hands" className="contributer-image"/>
            <p className="stats-title">{stats.contributers} {contributer}!</p>
          </div>
          <div className="stats-card">
            <img src={linesPhoto} alt="illustration of lines of code" className="lines-image"/>
            <p className="stats-title">{stats.lines} lines of code</p>
          </div>
          <div className="stats-card">
            <img src={updatesPhoto} alt="illustration of updates" className="updates-image"/>
            <p className="stats-title">{stats.updates} updates!</p>
          </div>
          <div className="stats-card">
            <img src={hourPhoto} alt="illustration of a hourglass" className="hour-image"/>
            <p className="stats-title">Roughly {stats.hours} hours spent</p>
          </div>
        </div>
      );
    } else {
      return (
        <div className="stats-container">
          <p className="stats-title">Nothing at this time</p>
        </div>

      ); 
    }
  }

  
  displayCategories = (categories) => {
    let display = null;
    if (categories) {
      display = categories.map((category, index)=> {
        return (
          <button key={index} className="display-category">{category}</button>
        );
      });
    }
    return display;
  }
  
  logInCheck = (client) => {
    const value = Object.keys(client).length;
    if ( value === 0 ) {
      return <Redirect to='/client-login'/>;
    }
  }
  
  render() {
    const redirect = this.logInCheck(this.props.client);
    let clientsProblem = {title: '', body: ''};
    let categories = '';
    if (this.props.clientsProblem) {
      clientsProblem = this.props.clientsProblem;
      categories = this.displayCategories(this.props.clientsProblem.categories);
    }
    const displayStats = this.displayStats(this.props.repoStats);
    

    return (
      <div className="frame-container">
        {redirect}
        <div className="problem">
          <p className="section-title">Heres your problem</p>
          <div className="problem-container">
            <p className="problem-title">{clientsProblem.title}</p>
            <p className="problem-body">{clientsProblem.body}</p>
            <div className="category-tags">
              {categories}
            </div>
          </div>
        </div>
        <div className="stats">
          <div className="stats-section">
            <p className="stats-section-title">Heres whats happening</p>
            {displayStats}
          </div>
        </div>
      </div>
      
    );
  }
}

export const mapStateToProps = (state) => ({
  client: state.client,
  clientsProblem: state.clientsProblem,
  repoStats: state.repoStats
});

export const mapDispatchToProps = (dispatch) => ({
  signInClient: (client) => dispatch(signInClient(client)),
  addRepoStats: (stats) => dispatch(addRepoStats(stats))
});

ClientProfile.propTypes = {
  clientsProblem: PropTypes.object,
  client: PropTypes.object,
  repoStats: PropTypes.bool,
  addRepoStats: PropTypes.func,
  signInClient: PropTypes.func,
  history: PropTypes.object
};

export default connect(mapStateToProps, mapDispatchToProps)(ClientProfile);