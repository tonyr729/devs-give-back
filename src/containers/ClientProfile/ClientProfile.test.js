import { ClientProfile } from './ClientProfile';
import { shallow } from 'enzyme';
import React from 'react';
import { Redirect } from 'react-router-dom';

describe('ClientProfile', () => {
  it('should match snapshot', () => {
    const mockProps = {
      client: {
        name: 'Tony'
      }
    }
    const wrapper = shallow(<ClientProfile {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });
  describe('logInCheck', () => {
    
    it('should redirect if there is no client', () => {
      const mockProps = {
        client: {}
      };
      
      const wrapper = shallow(<ClientProfile {...mockProps} />)

      const actual = wrapper.instance().logInCheck(mockProps)
      expect(actual).toEqual(<Redirect to='/client-login'/>)
    });
  });
});