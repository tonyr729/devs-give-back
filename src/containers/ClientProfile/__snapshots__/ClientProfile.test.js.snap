// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientProfile should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ClientProfile
    client={
        Object {
            "name": "Tony",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <div
          className="header"
>
          <div
                    className="header-container"
          >
                    <p>
                              Tony
                    </p>
                    <img
                              alt="user profile picture"
                              className="profile-picture"
                              src={undefined}
                    />
                    <button
                              className="signout-button"
                              onClick={[Function]}
                    >
                              Sign Out
                    </button>
          </div>
</div>,
        <div
          className="problem"
>
          <div
                    className="problem-container"
          >
                    <p
                              className="section-title"
                    >
                              Heres your problem
                    </p>
                    <p
                              className="problem-title"
                    />
                    <p
                              className="problem-body"
                    />
                    <div
                              className="category-tags"
                    />
          </div>
</div>,
        <div
          className="stats"
>
          <div
                    className="stats-section"
          >
                    <p
                              className="section-title"
                    >
                              Heres whats happening
                    </p>
                    <div
                              className="stats-container"
                    >
                              <p
                                        className="stats-title"
                              >
                                        Nothing at this time :(
                              </p>
                    </div>
          </div>
</div>,
      ],
      "className": "frame-container",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="header-container"
>
            <p>
                        Tony
            </p>
            <img
                        alt="user profile picture"
                        className="profile-picture"
                        src={undefined}
            />
            <button
                        className="signout-button"
                        onClick={[Function]}
            >
                        Sign Out
            </button>
</div>,
          "className": "header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Tony
</p>,
              <img
                alt="user profile picture"
                className="profile-picture"
                src={undefined}
/>,
              <button
                className="signout-button"
                onClick={[Function]}
>
                Sign Out
</button>,
            ],
            "className": "header-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Tony",
              },
              "ref": null,
              "rendered": "Tony",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "user profile picture",
                "className": "profile-picture",
                "src": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Out",
                "className": "signout-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Sign Out",
              "type": "button",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="problem-container"
>
            <p
                        className="section-title"
            >
                        Heres your problem
            </p>
            <p
                        className="problem-title"
            />
            <p
                        className="problem-body"
            />
            <div
                        className="category-tags"
            />
</div>,
          "className": "problem",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="section-title"
>
                Heres your problem
</p>,
              <p
                className="problem-title"
/>,
              <p
                className="problem-body"
/>,
              <div
                className="category-tags"
/>,
            ],
            "className": "problem-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Heres your problem",
                "className": "section-title",
              },
              "ref": null,
              "rendered": "Heres your problem",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "problem-title",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "problem-body",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "category-tags",
              },
              "ref": null,
              "rendered": "",
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="stats-section"
>
            <p
                        className="section-title"
            >
                        Heres whats happening
            </p>
            <div
                        className="stats-container"
            >
                        <p
                                    className="stats-title"
                        >
                                    Nothing at this time :(
                        </p>
            </div>
</div>,
          "className": "stats",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="section-title"
>
                Heres whats happening
</p>,
              <div
                className="stats-container"
>
                <p
                                className="stats-title"
                >
                                Nothing at this time :(
                </p>
</div>,
            ],
            "className": "stats-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Heres whats happening",
                "className": "section-title",
              },
              "ref": null,
              "rendered": "Heres whats happening",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p
                  className="stats-title"
>
                  Nothing at this time :(
</p>,
                "className": "stats-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Nothing at this time :(",
                  "className": "stats-title",
                },
                "ref": null,
                "rendered": "Nothing at this time :(",
                "type": "p",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <div
            className="header"
>
            <div
                        className="header-container"
            >
                        <p>
                                    Tony
                        </p>
                        <img
                                    alt="user profile picture"
                                    className="profile-picture"
                                    src={undefined}
                        />
                        <button
                                    className="signout-button"
                                    onClick={[Function]}
                        >
                                    Sign Out
                        </button>
            </div>
</div>,
          <div
            className="problem"
>
            <div
                        className="problem-container"
            >
                        <p
                                    className="section-title"
                        >
                                    Heres your problem
                        </p>
                        <p
                                    className="problem-title"
                        />
                        <p
                                    className="problem-body"
                        />
                        <div
                                    className="category-tags"
                        />
            </div>
</div>,
          <div
            className="stats"
>
            <div
                        className="stats-section"
            >
                        <p
                                    className="section-title"
                        >
                                    Heres whats happening
                        </p>
                        <div
                                    className="stats-container"
                        >
                                    <p
                                                className="stats-title"
                                    >
                                                Nothing at this time :(
                                    </p>
                        </div>
            </div>
</div>,
        ],
        "className": "frame-container",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="header-container"
>
              <p>
                            Tony
              </p>
              <img
                            alt="user profile picture"
                            className="profile-picture"
                            src={undefined}
              />
              <button
                            className="signout-button"
                            onClick={[Function]}
              >
                            Sign Out
              </button>
</div>,
            "className": "header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Tony
</p>,
                <img
                  alt="user profile picture"
                  className="profile-picture"
                  src={undefined}
/>,
                <button
                  className="signout-button"
                  onClick={[Function]}
>
                  Sign Out
</button>,
              ],
              "className": "header-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Tony",
                },
                "ref": null,
                "rendered": "Tony",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "user profile picture",
                  "className": "profile-picture",
                  "src": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign Out",
                  "className": "signout-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Sign Out",
                "type": "button",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="problem-container"
>
              <p
                            className="section-title"
              >
                            Heres your problem
              </p>
              <p
                            className="problem-title"
              />
              <p
                            className="problem-body"
              />
              <div
                            className="category-tags"
              />
</div>,
            "className": "problem",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="section-title"
>
                  Heres your problem
</p>,
                <p
                  className="problem-title"
/>,
                <p
                  className="problem-body"
/>,
                <div
                  className="category-tags"
/>,
              ],
              "className": "problem-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Heres your problem",
                  "className": "section-title",
                },
                "ref": null,
                "rendered": "Heres your problem",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "className": "problem-title",
                },
                "ref": null,
                "rendered": "",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "className": "problem-body",
                },
                "ref": null,
                "rendered": "",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "className": "category-tags",
                },
                "ref": null,
                "rendered": "",
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="stats-section"
>
              <p
                            className="section-title"
              >
                            Heres whats happening
              </p>
              <div
                            className="stats-container"
              >
                            <p
                                          className="stats-title"
                            >
                                          Nothing at this time :(
                            </p>
              </div>
</div>,
            "className": "stats",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="section-title"
>
                  Heres whats happening
</p>,
                <div
                  className="stats-container"
>
                  <p
                                    className="stats-title"
                  >
                                    Nothing at this time :(
                  </p>
</div>,
              ],
              "className": "stats-section",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Heres whats happening",
                  "className": "section-title",
                },
                "ref": null,
                "rendered": "Heres whats happening",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <p
                    className="stats-title"
>
                    Nothing at this time :(
</p>,
                  "className": "stats-container",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Nothing at this time :(",
                    "className": "stats-title",
                  },
                  "ref": null,
                  "rendered": "Nothing at this time :(",
                  "type": "p",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
